content/developer/embeds/index.md

```diff
@@ -0,0 +1,100 @@
+    * height: Height of the resource (e.g. video height).
+    * width: Width of the resource (e.g. video width).
+    * attributes: An array of arbitrary associated data (e.g. a video ID).
+* `renderData` - Embed markup generated for inclusion on a page is expected to be returned by this method. The data array parameter should be the same as was returned by the `matchUrl` method in the class.
+
+### Registering the Embed Component
+
+Once an embed class has been created, it must be registered with the embed manager. The embed manager is the centralized handler of what embed classes are called, based on a URL. New embeds must be added with the `EmbedManager::addEmbed` method, called on the shared instance. This requires hooking early into Vanilla's setup, but after the dependency injection container has been initialized. The `container_init` event can be used to perform this action.
+
+```php
+public function container_init_handler(Container $dic) {
+    $dic->rule(Vanilla\Embeds\EmbedManager::class)
+        ->addCall('addEmbed', [new Reference(MyEmbed::class)])
+}
+```
+
+After the embed class has been registered, subsequent calls to the embed manager will use the class for URL scraping and markup generation. To test, you can use Vanilla's API v2 media endpoint to scrape a URL that should be handled by the new component.
+
+## Client-Side Components
+
+Client-side rendering is performed with JavaScript. The implementation of a client-side embed renderer is straightforward, primarily powered by a single renderer function.
+
+Vanilla's `registerEmbed` function is required to register the new renderer. It must be imported from `@dashboard/embeds`. In addition, the `IEmbedData` interface is also required for  constructing a proper render function signature. To import these two requirements, you may include the following:
+
+```javascript
+import { registerEmbed, IEmbedData } from "@dashboard/embeds";
+```
+
+The renderer is expected to take two parameters: `element` and `data`. The `element` parameter is the HTML element that will contain the rendered embed, while `data` is expected to be an instance of `IEmbedData`, containing the data returned from a server-side embed component's `matchUrl` method. This function returns no value. It is expected to act on `element`.
+
+```javascript
+export async function myRenderer(element: HTMLElement, data: IEmbedData)
+```
+
+Lastly, a call to `registerEmbed` is necessary to ensure when an embed of the target type is encountered, the renderer function is called.
+
+```javascript
+registerEmbed("custom", myRenderer);
+```
+
+All the pieces can be seen coming together in the image embed renderer.
+
+```javascript
+import { registerEmbed, IEmbedData, FOCUS_CLASS } from "@dashboard/embeds";
+
+// Setup image embeds.
+registerEmbed("image", imageRenderer);
+
+/**
+ * Render an image embed in the editor.
+ */
+export async function imageRenderer(element: HTMLElement, data: IEmbedData) {
```

Comment:
This example is of an async function that doesn't actually do anything asynchronously. I think we should offer a plain synchronous example like this, as well as one that does stuff asynchronously.